tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1093,6 +1122,53 @@ private[optimizer] abstract class OptimizerCore(
       keepOnlySideEffects(tree)
   }
 
+  /** Finishes the bindings and statements followed by a result to get a
+   *  normal [[Tree]].
+   *  This method must not be called more than once per `BindingOrStat` and
+   *  per translation.
+   *  By "per translation", we mean in an alternative path through
+   *  `tryOrRollback`. It could still be called several times as long as
+   *  it is once in the 'try' part and once in the 'fallback' part.
+   */
+  private def finishTransformBindings(bindingsAndStats: List[BindingOrStat],
+      result: Tree): Tree = {
+    bindingsAndStats.foldRight(result) {
+      case (Left(PreTransBinding(localDef, value)), innerBody) =>
+        implicit val pos = value.pos
+
+        val LocalDef(tpe, mutable, replacement) = localDef
+
+        val (name, originalName, used) = (replacement: @unchecked) match {
+          case ReplaceWithVarRef(name, originalName, used, _) =>
+            (name, originalName, used)
+          case ReplaceWithRecordVarRef(name, originalName, _, used, _) =>
+            (name, originalName, used)
+        }
+
+        if (used.value) {
+          val ident = Ident(name, originalName)
+          val varDef = resolveLocalDef(value) match {
+            case PreTransRecordTree(valueTree, valueTpe, cancelFun) =>
+              val recordType = valueTree.tpe.asInstanceOf[RecordType]
+              if (!isImmutableType(recordType))
```

Comment:
We can inline them (as in, alias them through `LocalDef`s) but we can never *copy* them, as in an `Assign` or a `VarDef`, otherwise they lose their identity. So if you change the copy, the original is not modified, and conversely.